File: backend/app/v1/schemas/department.py
"""
Pydantic schemas for Department-related data transfer.
"""

import uuid

from pydantic import BaseModel, ConfigDict


class DepartmentBase(BaseModel):
    """
    Base schema for Department data with shared attributes.
    """

    name: str
    description: str | None = None


class DepartmentCreate(DepartmentBase):
    """
    Schema for creating a new Department.
    """

    pass


class DepartmentUpdate(BaseModel):
    """
    Schema for updating an existing Department.
    """

    name: str | None = None
    description: str | None = None


class DepartmentRead(DepartmentBase):
    """
    Schema for reading Department data, including database-generated fields.
    """

    id: uuid.UUID

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/department.ts
import type { DepartmentCreateFormValues, DepartmentUpdateFormValues } from "@/schemas/department";

/**
 * Shared fields for a department.
 */
export interface DepartmentBase {
  id: string;
  name: string;
  description?: string;
}

/**
 * Payload for creating a new department.
 */
export type DepartmentCreate = DepartmentCreateFormValues

/**
 * Payload for updating an existing department.
 */
export type DepartmentUpdate = DepartmentUpdateFormValues

/**
 * Department returned from read operations.
 */
export type DepartmentRead = DepartmentBase
